src/app/services/apiservice.ts

```diff
@@ -56,17 +73,45 @@ export class ApiService {
         });
     });
   }
-  listpoll(post) {
-    return new Promise((resolve, reject) => {
-      this.http
-        .get(`${environment["apiBase"]}list_polls`)
-        .subscribe(data => {
-          if (data["error"] && data["error"] === 1) {
-            reject(data);
-          } else {
-            resolve(data);
-          }
-        });
-    });
+
+  listpolls() {
+    this.role = localStorage.getItem("role").toLowerCase();
+    if (this.role == "admin") {
```

Comment:
```
 if (this.role == "admin") {
      return this.http.get(
        `${environment["apiBase"]}list_polls`,
        this.httpOptions
      );
    } else {
      return this.http.get(
        `${environment["apiBase"]}list_poll`,
        this.httpOptions
      );
    }
```

why this if condition. Didn't understand.

Reply:
because sir in this if admin is login then appear a poll list but when guest login then it show only guest poll